src/rate-limiter.ts

```diff
@@ -8,10 +10,61 @@ interface CaptureResponse {
 }
 
 export class RateLimiter {
```

Comment:
should we share the token per second similarity with the mutation rate limiter?
might at least reduce the amount we add to the bundle

Reply:
Not sure if it would actually reduce it tbh. The code is somewhat different so would need abstracting to make reusable which in turn would increase bundle size 😅

Reply:
it would be so cool to move all of this into the posthog-js-lite repo so we really had "stripped back" and "full featured" off the same base code

Reply:
Yeah thats a dream I'm not sure we will ever achieve...

Reply:
I'll just keep mentioning it until its more annoying to not do it 🤣